dmoj/cptbox/landlock_helpers.cpp

```diff
@@ -0,0 +1,42 @@
+#ifndef __FreeBSD__
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <string>
+
+#include "landlock_helpers.h"
+
+int landlock_add_path(struct landlock_path_beneath_attr &rule, const int ruleset_fd, const char *path) {
+    rule.parent_fd = open(path, O_PATH | O_CLOEXEC);
+
+    if (rule.parent_fd < 0) {
+        if (errno == ENOENT)
+            goto close_fd;  // missing files are ignored
```

Comment:
Nit: parens even around single-statement branches / loops / etc. I know not all cptbox code is written like this, but all new code should be.